index.d.ts

```diff
@@ -0,0 +1,1100 @@
+// Type definitions for [~THE LIBRARY NAME~] [~OPTIONAL VERSION NUMBER~]
+// Project: [~THE PROJECT NAME~]
+// Definitions by: [~YOUR NAME~] <[~A URL FOR YOU~]>
+
+export as namespace Draggable;
+
+// type definitions
+export type DraggableContainer = HTMLElement[] | NodeList | HTMLElement
+
+type DraggableEventName = 'draggable:initialized' | 'draggable:destroy' | string
+type DragEventName = 'drag:start' | 'drag:move' | 'drag:over' | 'drag:over:container' | 'drag:out' | 'drag:out:container' | 'drag:stop' | 'drag:pressure' | string
+type SortEventName = 'sortable:start' | 'sortable:sorted' | 'sortable:stop' | string
+type MirrorEventName = 'mirror:created' | 'mirror:attached' | 'mirror:move' | 'mirror:destroy' | string
+
+// plugin
+export class DraggablePlugin {
+  draggable: Draggable
+
+  constructor(draggable: Draggable)
+
+  attach(): void
+
+  detach(): void
+
+  getOptions(): Object
+}
+
+// sensor
+export class DraggableSensor {
+
+  /**
+   * Current containers
+   * @property containers
+   * @type {HTMLElement[]}
+   */
+  containers: HTMLElement[]
+
+  /**
+   * Current options
+   * @property options
+   * @type {Object}
+   */
+  options: Object
+
+  /**
+   * Current drag state
+   * @property dragging
+   * @type {Boolean}
+   */
+  dragging: boolean
+
+  /**
+   * Current container
+   * @property currentContainer
+   * @type {HTMLElement}
+   */
+  currentContainer: HTMLElement
+
+  constructor(containers: HTMLElement[], options: Object)
+
+  attach(): DraggableSensor
```

Comment:
I don't think it actually matters whether this returns a `DraggableSensor`, this might be a legit case for an any, or better a generic.

```
class Sensor<AttachT = undefined, DetachT = undefined> {
 // other stuff
 attach(): AttachT
 detach(): DetachT
}
```

You should be able to then extend it with the types you care about when you make your own classes, eg.

```
class MySensor extends Sensor<Something, OtherThing> {}
```

or leave it and then it will expect no return value.